kornia/onnx/sequential.py

```diff
@@ -0,0 +1,149 @@
+        """
+        # TODO: maybe another logic for versioning.
+        if target_ir_version is None:
+            target_ir_version = 9
+        if target_opset_version is None:
+            target_opset_version = 17
+
+        op_list = []
+        for op in args:
+            op = super()._onnx_version_conversion(
+                op, target_ir_version=target_ir_version, target_opset_version=target_opset_version
+            )
+            op_list.append(op)
+        return op_list
+
+    def _combine(
+        self,
+        *args: List["onnx.ModelProto"],  # type:ignore
+        io_maps: Optional[List[Tuple[str, str]]] = None,
+    ) -> "onnx.ModelProto":  # type:ignore
+        """Combine the provided ONNX models into a single ONNX graph. Optionally, map inputs and outputs between
+        operators using the `io_map`.
+
+        Args:
+            io_maps:
+                A list of list of tuples representing input-output mappings for combining the models.
+                Example: [[(model1_output_name, model2_input_name)], [(model2_output_name, model3_input_name)]].
+
+        Returns:
+            onnx.ModelProto: The combined ONNX model as a single ONNX graph.
+        """
+        if len(args) == 0:
+            raise ValueError("No operators found.")
+
+        combined_op = args[0]
+        combined_op = onnx.compose.add_prefix(combined_op, prefix=f"K{str(0).zfill(2)}-")  # type:ignore
+
+        for i, op in enumerate(args[1:]):
+            next_op = onnx.compose.add_prefix(op, prefix=f"K{str(i + 1).zfill(2)}-")  # type:ignore
+            if io_maps is None:
+                io_map = [(f"K{str(i).zfill(2)}-output", f"K{str(i + 1).zfill(2)}-input")]
+            else:
+                io_map = [(f"K{str(i).zfill(2)}-{it[0]}", f"K{str(i + 1).zfill(2)}-{it[1]}") for it in io_maps[i]]
+            combined_op = onnx.compose.merge_models(combined_op, next_op, io_map=io_map)  # type:ignore
+
+        return combined_op
+
+    def export(self, file_path: str, **kwargs: Any) -> None:
+        return super()._export(self._combined_op, file_path, **kwargs)
+
+    def add_metadata(self, additional_metadata: List[Tuple[str]] = []) -> "onnx.ModelProto":  # type:ignore
+        return super()._add_metadata(self._combined_op, additional_metadata)
+
+    async def run(self, *inputs: "np.ndarray") -> List["np.ndarray"]:  # type:ignore
+        """Perform inference asynchronously using the ONNX model.
+
+        Args:
+            *inputs: Inputs to the ONNX model. The number of inputs must match the
+                expected inputs of the session.
+
+        Returns:
+            List: The outputs from the ONNX model inference.
+        """
+        loop = asyncio.get_event_loop()
+        ort_inputs = self._session.get_inputs()
+        ort_input_values = {ort_inputs[i].name: inputs[i] for i in range(len(ort_inputs))}
+        outputs = await loop.run_in_executor(None, self._session.run, None, ort_input_values)
```

Comment:
onnxruntime has already its own async method. https://onnxruntime.ai/docs/api/python/api_summary.html#onnxruntime.InferenceSession.run_async

I would leave this method run as synchronous